client/asset/btc/btc.go

```diff
@@ -1362,6 +1407,43 @@ func (btc *baseWallet) feeRate(_ RawRequester, confTarget uint64) (uint64, error
 	return uint64(dex.IntDivUp(int64(satPerKB), 1000)), nil
 }
 
+// externalFeeEstimator gets the fee rate from the external API
+func externalFeeEstimator(ctx context.Context, net dex.Network, nb uint64) (float64, error) {
+	var url string
+	if net == dex.Testnet {
+		url = testnetExternalApiUrl
+	} else {
+		url = externalApiUrl
+	}
+	url = url + "v1/fees/recommended"
+	ctx, cancel := context.WithTimeout(ctx, 4*time.Second)
+	defer cancel()
+	r, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+
+	if err != nil {
+		return 0, err
+	}
+	httpResponse, err := client.Do(r)
+	if err != nil {
+		return 0, err
+	}
+	var resp map[string]uint
```

Comment:
Please used sized integer types to guarantee identical data between 32-bit and 64-bit platforms.